tests/ui/pages/catalog/catalog-page.ts

```diff
@@ -0,0 +1,56 @@
+import InputField from '../../elements/input-field';
+import BasePage from '../base-page';
+
+const SELECTORS = {
+  filterWithSelect: filterName => `#select-label-id:has-text('${filterName}')`,
+  filterWithInput: filterName => `label:text-is("${filterName}") >> ..`,
+  searchBar: `[placeholder="Search"]`,
+  listOfFilters: `[role="listbox"]`,
+  filterOption: `[role="option"]`,
+  filterWithInputOption: `[role="presentation"]`,
+  listItemName: name => `a:has-text('${name}')`,
+  tab: name => `[role="tab"]:has-text('${name}')`,
+};
+export default class CatalogPage extends BasePage {
+  async openFilterWithSelect(filterName: string) {
+    const filterElement = this.page.locator(SELECTORS.filterWithSelect(filterName)).locator('..');
```

Comment:
Please modify `filterWithSelect` selector, so that it will wont be needed to call  `.locator('..');`

For example, see selector `filterWithInput`